src/server/lib/oauth/client.js

```diff
@@ -238,8 +239,15 @@ async function getOAuth2 (provider, accessToken, results) {
     url = prepareProfileUrl({ provider, url, results })
   }
 
+  /** Dropbox requires POST instead of GET
+   * Read more: https://www.dropbox.com/developers/reference/auth-types#user
+   */
+  if (provider.id === 'dropbox') {
+    httpMethod = 'POST'
+  }
```

Comment:
So I really push against provider-specific handling, unless it is absolutely necessary. At the same time, I try to keep the provider options surface minimal. When #1698 is done, we will be able to loan some of their API, and I can see they have something for this exact situation: https://github.com/panva/node-openid-client/blob/main/docs/README.md#clientuserinfoaccesstoken-options, where you can tell the library what method to use for the `userinfo` endpoint. I think we could go with something like this, so this change would also be useful for future providers that support POST in this case. What do you think?

Reply:
@balazsorban44 
I couldn't see any other way around this currently. But, agree that the `method` option in the `userinfo` would be perfect. I'd be happen to update once this is supported.